server/src/controllers/blogController.js

```diff
@@ -0,0 +1,124 @@
+    if (!img) return Response.error(res, 500, "Failed to upload photo");
+    try {
+      const blog = await Blog.create({
+        title: req.body.title,
+        content: req.body.content,
+        blogImage: img.url,
+      });
+      Response.success(res, 201, "Sucessfully Created the blog", blog);
+    } catch (err) {
+      if (err.code === 11000)
+        return Response.error(res, 400, "Duplicate field value entered");
+      Response.error(res, 500, err);
+    }
+  }
+
+  static async getBlog(req, res) {
+    try {
+      const blog = await Blog.findOne({ _id: req.params.id }).populate(
+        "comments"
+      );
+      if (blog === null) return Response.error(res, 404, "Blog not found");
+      return Response.success(res, 200, "Sucessfully Retrieved the blog", blog);
+    } catch (err) {
+      Response.error(res, 404, "Blog not found");
+    }
+  }
+
+  static async updateBlog(req, res) {
+    const img = await uploadToCloud(req.file);
+    if (!img) return Response.error(res, 500, "Failed to upload photo");
+    try {
+      const blogUpdated = await Blog.findOneAndUpdate(
+        req.params.id,
+        {
+          title: req.body.title,
+          content: req.body.content,
+          blogImage: img.url,
+        },
+        { new: true }
+      );
+      if (blogUpdated === null)
+        return Response.error(res, 404, "Blog not found");
+      return Response.success(
+        res,
+        200,
+        "Sucessfully Updated the blog",
+        blogUpdated
+      );
+    } catch (error) {
+      return Response.error(res, 404, "Blog not found");
+    }
+  }
+
+  static async deleteBlog(req, res) {
+    try {
+      const deleteBlog = await Blog.findByIdAndRemove(req.params.id);
+      if (deleteBlog === null)
+        return Response.error(res, 404, "Blog not found");
+      return Response.success(
+        res,
+        200,
+        "Sucessfully deleted the blog",
+        deleteBlog
+      );
+    } catch (error) {
+      return Response.error(res, 404, "Blog not found");
+    }
+  }
+
+  static async addComment(req, res) {
+    const blogId = req.params.id;
+    if (!blogId) return Response.error(res, 400, "Please provide the blogId");
+    const blog = await Blog.findById(blogId);
+    if (!blog) return Response.success(res, 404, "Blog Not Found");
+    try {
+      const newComment = new Comment({ comment: req.body.comment });
+      await newComment.save();
+      blog.comments.push(newComment);
+      blog.commentCounts += 1;
+      await blog.save();
+      return Response.success(
+        res,
+        201,
+        "Comment sucessfully Created",
+        newComment
+      );
+    } catch (error) {
+      return Response.error(res, 500, "something went wrong");
+    }
+  }
+
+  static async addLike(req, res) {
+    const blogId = req.params.id;
+    if (!blogId) return Response.error(res, 400, "Please provide the blogId");
+    const blog = await Blog.findById(blogId);
```

Comment:
Kindly put this in try scope

Reply:
Yes I am going to implement try catch scope but with checking if the user have entered the blog id is not necessary because it is telling me in the postman that the route does not exists. what I will try to implement is to have a route that will display that the route a user is trying to reach does not exist.

Reply:
Okay if that is the case, would remove those codes from the controller and do middleware for DB validation. By saying DB validation I mean having a middleware that will be checking those conditions.